### ISSUE TITLE
Position and set size not narrated for gridcell in treegrid

### ISSUE THREAD

**Issue by adamsamec** [nvaccess/nvda#15818]
### Steps to reproduce:

1. Turn on NVDA.
2. Visit this [W3C tree grid cell navigation example #1][cell example].
3. Using the Tab key, navigate to the inbox example tree grid element located under the "Example" heading.
4. Using the Down or Up arrow key, navigate by rows in the tree grid, and observe the narration while navigating from one row to another.
5. Visit this [W3C tree grid row navigation example #2][row example].
6. Using the Tab key, navigate to the inbox example tree grid element located under the "Example" heading.
7. Using the Down or Up arrow key, navigate by rows in the tree grid, and observe the narration while navigating from one row to another.

### Actual behavior:

In the example #1, when nnavigating in the tree grid with the Down or Up arrow key, NVDA narrates the row and colun coordinates, but not the position of the row and the set size relative to the current tree level. Compare this with the example #2, where NVDA narrates the position and set size correctly, e.g. "2 of 3". The narration is missing regardless of whether the aria-posinset and aria-setsize are present on the row or not.

### Expected behavior:

The position and set size should be narrated even when role="gridcell" element is focused, inheriting the data from the parent role="row" element, in the same manner how JAWS does.

#### NVDA installed/portable/running from source:

Installed.

#### NVDA version:

NVDA 2023.3.0.29780.

#### Windows version:

Windows 11, version 21H2 (OS Build 22621.2715)

#### Name and version of other software in use when reproducing the issue:

Google Chrome, version 119.0.6045.160.

#### Does the issue still occur after restarting your computer?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes.

[cell example]: https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/?cell=force
[row example]: https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/?cell

**Comment by Adriani90** [comment]
Is this also reproducible in Firefox?

**Comment by adamsamec** [comment]
Yes, it is reproducable in Mozilla Firefox. I have updated the issue.

**Comment by Adriani90** [comment]
cc: @jcsteh since it affects Firefox as well.

**Comment by SaschaCowley** [comment]
@adamsamec could you provide a concrete example of the type of speech and/or braille output you'd expect when navigating a treegrid by cell?

Can anyone provide any examples of this authoring pattern in the wild?
